spark/src/main/scala/org/apache/spark/sql/delta/SnapshotManagement.scala

```diff
@@ -253,6 +239,12 @@ trait SnapshotManagement { self: DeltaLog =>
     )
   }
 
+  private def createLogSegment(previousSnapshot: Snapshot): Option[LogSegment] = {
```

Comment:
This is just a convenience overload to create a new log segment, using the previous snapshot as starting point, right? Otherwise nothing special?
(maybe worth a quick doc comment)